test-network-function/lifecycle/suite.go

```diff
@@ -293,14 +293,20 @@ func testPodsRecreation(env *config.TestEnvironment) {
 	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestPodRecreationIdentifier)
 	ginkgo.It(testID, func() {
 		ginkgo.By("Testing node draining effect of deployment")
-		ginkgo.By(fmt.Sprintf("test deployment in namespace %s", env.NameSpaceUnderTest))
-		deployments, notReadyDeployments = getDeployments(env.NameSpaceUnderTest)
-		if len(deployments) == 0 {
-			return
+		ginkgo.By(fmt.Sprintf("test deployment in namespace %s", env.NameSpacesUnderTest))
+		for _, ns := range env.NameSpacesUnderTest {
+			var dps dp.DeploymentMap
+			dps, notReadyDeployments = getDeployments(ns)
+			for dpKey, dp := range dps {
+				deployments[dpKey] = dp
```

Comment:
Based on the failure in the unit test, we need to initialize deployments in its declaration (line 290 in the file). This was not an issue previously, because we got it from getDeployments().

Reply:
fixed